frontend/src/features/projects/dispose/ProjectDisposeLayout.tsx

```diff
@@ -44,19 +47,44 @@ const ProjectDisposeLayout = ({ match, location }: { match: Match; location: Loc
   const projectNumber = queryString.parse(query).projectNumber;
   const historyReplace = history.replace;
 
-  // TODO: UX flow - since this is a pre draft state hitting next does not step through the stepper as once they create the project then they are in 'Draft'
-  // May be something we want to change
-  const onNext = async () => {
-    if (!projectNumber) {
-      await handleSubmit();
+  const updateProjectStatus = (
+    project: IProject,
+    nextStepId: number,
+    workflowStatusCode?: string,
+  ) => {
+    if (project.statusId === currentStatus.id) {
+      return dispatch(updateWorkflowStatus(project, nextStepId, workflowStatusCode) as any).then(
+        (project: IProject) => {
+          goToNextStep(project);
+          if (nextStepId === ReviewWorkflowStatus.PropertyReview) {
+            history.push('/project/completed');
+          }
+          return project;
+        },
+      );
     }
-    formikRef.current?.submitForm().then((value: any) => {
-      if (!formikRef?.current?.errors || !Object.keys(formikRef?.current?.errors).length) {
-        // do not go to the next step if the form has validation errors.
-        const hasNextStep = nextStep();
-        if (!hasNextStep) {
-          history.push('/project/completed');
+    goToNextStep(project);
+    return Promise.resolve(project);
+  };
+
+  const onNext = () => {
+    formikRef.current?.submitForm().then(() => {
+      const values = formikRef?.current?.values;
+      const errors = formikRef?.current?.errors;
+      // do not go to the next step if the form has validation errors.
+      if (errors === undefined || !Object.keys(errors).length) {
+        let nextStepId = getNextStep(currentStatus)?.id;
+        let workflowStatusCode: string | undefined = undefined;
+        if (nextStepId === undefined) {
+          nextStepId = ReviewWorkflowStatus.PropertyReview;
+          workflowStatusCode = 'ACCESS-DISPOSAL';
         }
+
+        addOrUpdateProject(values, formikRef).then((project: IProject) =>
+          updateProjectStatus(project, nextStepId!, workflowStatusCode).then((project: IProject) =>
+            dispatch(fetchProject(project.projectNumber)),
```

Comment:
Why do you fetch the project at this point?  You should have an updated copy from the prior actions.

Reply:
Last time I checked the create/update apis were not correctly returning the updated projects from the database, they were just returning a copy of the passed in object. I brought this up in a code review at the start of this sprint. Has this been fixed?

Reply:
If they're not returning the correct data, they need to be updated.  Should be as simple as
- return correct object
- make sure the right data is included in that object